app/models/annotation_text.rb

```diff
@@ -3,15 +3,30 @@ class AnnotationText < ApplicationRecord
   belongs_to :creator, class_name: 'User', foreign_key: :creator_id
   belongs_to :last_editor, class_name: 'User', foreign_key: :last_editor_id, optional: true
 
+  after_update :update_mark_deductions
+
   # An AnnotationText has many Annotations that are destroyed when an
   # AnnotationText is destroyed.
   has_many :annotations, dependent: :destroy
 
   belongs_to :annotation_category, optional: true, counter_cache: true
   validates_associated :annotation_category, on: :create
 
+  validates_numericality_of :deduction,
```

Comment:
You might need to add an `if:` option to guard when the `deduction` or the associated `flexible_criterion` is `nil`, I can't remember how these validates work.